### PR TITLE
[ci skip] Update caching development mode default status docs

### PR DESCRIPTION
**Pull request by starwels**
### Motivation / Background

This Pull Request has been created because the default state of caching on a fresh rails app is wrong in the docs

### Detail

`Rails.application.config.action_controller.perform_caching` returns false in a fresh created rails app

### Checklist

Before submitting the PR make sure the following are checked:

* [x] This Pull Request is related to one change. Unrelated changes should be opened in separate PRs.
* [x] Commit message has a detailed description of what changed and why. If this PR fixes a related issue include it in the commit message. Ex: `[Fix #issue-number]`
* [x] Tests are added or updated if you fix a bug or add a feature.
* [x] CHANGELOG files are updated for the changed libraries if there is a behavior change or additional feature. Minor bug fixes and documentation changes should not be included.

### CODE REVIEW

**Comment by p8** [comment]
Are you sure?
By default `:memory_store` is enabled:
https://github.com/rails/rails/blob/5f71da152142b2abd523eac17ed0058690edd744/railties/lib/rails/generators/rails/app/templates/config/environments/development.rb.tt#L33

ActionController caching is still disabled by default though.
See: https://github.com/rails/rails/pull/52474 and https://github.com/rails/rails/pull/52474

I think the wording can be improved though...

**Comment by p8** [comment]
We also need to change the following to make clear it's only for ActionController caching: https://github.com/rails/rails/blob/5f71da152142b2abd523eac17ed0058690edd744/railties/lib/rails/dev_caching.rb#L13-L19

File: guides/source/caching_with_rails.md [hunk 1]
```diff
@@ -721,7 +721,7 @@ response.strong_etag = response.body # => "618bbc92e2d35ea1945008b42799b0e7"
 Caching in Development
 ----------------------
 
-By default, caching is *enabled* in development mode with
+By default, caching is *disabled* in development mode with
 [`:memory_store`](#activesupport-cache-memorystore).
```
Comment (p8) [inline guides/source/caching_with_rails.md:725]:
```suggestion
By default, caching is *enabled* in development mode with
[`:memory_store`](#activesupport-cache-memorystore).
This doesn't apply to Action Controller caching, which is disabled
by default.
```

File: guides/source/caching_with_rails.md [hunk 2]
```diff
@@ -721,7 +721,7 @@ response.strong_etag = response.body # => "618bbc92e2d35ea1945008b42799b0e7"
...
 
 Rails also provides the rails command `dev:cache` to
```
Comment (p8) [inline guides/source/caching_with_rails.md:727]: This should probably be something like:

```
To enable Action Controller caching Rails provides the `bin/rails dev:cache` command.
```

**Comment by starwels** [comment]
> We also need to change the following to make clear it's only for ActionController caching:
> 
> https://github.com/rails/rails/blob/5f71da152142b2abd523eac17ed0058690edd744/railties/lib/rails/dev_caching.rb#L13-L19
> 
> And here:
> 
> https://github.com/rails/rails/blob/5f71da152142b2abd523eac17ed0058690edd744/railties/lib/rails/commands/dev/dev_command.rb#L8

Can I make those changes in this PR or should I open a new one just for the code change?

**Comment by p8** [comment]
@starwels I think that if we are going to change this, we might as well fix it in all occurences.
So yes, add it to this PR please.

File: railties/lib/rails/dev_caching.rb [hunk 3]
```diff
@@ -12,10 +12,10 @@ def enable_by_file
 
         if File.exist?(FILE)
           delete_cache_file
-          puts "Development mode is no longer being cached."
+          puts "Action Controller development mode is no longer being cached."
```
Comment (p8) [inline railties/lib/rails/dev_caching.rb:15]:
```suggestion
          puts "Action Controller caching disabled for development mode."
```

File: railties/lib/rails/dev_caching.rb [hunk 4]
```diff
@@ -12,10 +12,10 @@ def enable_by_file
...
         else
           create_cache_file
-          puts "Development mode is now being cached."
+          puts "Action Controller development mode is now being cached."
```
Comment (p8) [inline railties/lib/rails/dev_caching.rb:18]:
```suggestion
          puts "Action Controller caching enabled for development mode."
```

**Comment by p8** [comment]
Thanks @starwels !
Can you squash your commits?

**Review by p8 — APPROVED** [review]

**Pull request merged**